### ISSUE TITLE
Allow several pushes without pulls

### ISSUE THREAD

**Issue by grimmySwe** [ingydotnet/git-subrepo#107]
Current workflow is that when you perform a push to a subrepo you must perform a pull to update your local `.gitrepo` file (if you don't supply the --update flag).
```
git subrepo push bar
git subrepo pull bar       <== You must do this otherwise it will complain that you haven't updated.
modify bar
git subrepo push bar
```
Consider the possibility of always using the ancestor check to see if you are allowed to push. That way it will be more similar to the regular git workflow where you don't have to perform pull after each push.
```
git subrepo push bar
modify bar
git subrepo push bar
```

**Comment by grimmySwe** [comment]
Implemented a proposed solution on branch issue/107 (this also includes fixes for #106 and #108)

**Comment by ingydotnet** [comment]
Fixed by #110
